pkg/helper/ressource_block.go

```diff
@@ -0,0 +1,149 @@
+	)
+
+	var r Ressource
+	r.Ressource = ressource
+	r.Name = defaultName
+	r.StringValues = map[string]string{
+		"region":          defaultRegion,
+		"smallest_flavor": defaultSmallestFlavor,
+		"biggest_flavor":  defaultBiggestFlavor,
+	}
+	r.IntValues = map[string]int{
+		"min_instance_count": dafaultMinInstances,
+		"max_instance_count": defaultMaxInstances,
+	}
+
+	// RessourceOption functions
+	for _, opt := range opts {
+		opt(&r)
+	}
+
+	return &r
+}
+
+// Name value setter:
+//   - desc: concatenate function that set Ressource.Name then return Ressource
+//   - args: new name
+//   - return: pointer to Ressource
+func (r *Ressource) SetName(newName string) *Ressource {
+	r.Name = newName
+	return r
+}
+
+// String value setter:
+//   - desc: set/add key: value to the string values map of a Ressource then return the Ressource
+//   - args: key + value
+//   - return: pointer to Ressource
+func (p *Ressource) SetOneStringValue(key, value string) *Ressource {
+	p.StringValues[key] = value
+	return p
+}
+
+// Integer value setter:
+//   - desc: set/add key: value to the int values map of a Ressource then return the Ressource
+//   - args: key + value
+//   - return: pointer to Ressource
+func (p *Ressource) SetOneIntValue(key string, value int) *Ressource {
+	p.IntValues[key] = value
+	return p
+}
+
+// String values setter:
+//   - desc: set/add a map of key: value to the string values map of a Ressource then return the Ressource
+//   - args: map of key + value
+//   - return: pointer to Ressource
+func (p *Ressource) SetStringValues(stringMap map[string]string) *Ressource {
+	for key, value := range stringMap {
+		p.StringValues[key] = value
+	}
+	return p
+}
+
+// Integer values setter:
+//   - desc: set/add a map of key: value to the int values map of a Ressource then return the Ressource
+//   - args: map of key + value
+//   - return: pointer to Ressource
+func (p *Ressource) SetIntValues(intMap map[string]int) *Ressource {
+	for key, value := range intMap {
+		p.IntValues[key] = value
+	}
+	return p
+}
+
+// Ressource block
+//   - desc: chained function that stringify Ressource into a terraform block
+//   - args: none
+//   - return: string
+func (p *Ressource) String() string {
+	s := `ressource "` + p.Ressource + `" "` + p.Name + `" {
+	name = "` + p.Name + `"
+`
+	// check StringValues not empty
+	if len(p.StringValues) != 0 {
+		// sort StringValues keys
+		tmp := make([]string, 0, len(p.StringValues))
+		for k := range p.StringValues {
+			tmp = append(tmp, k)
+		}
+		sort.Strings(tmp)
+
+		// create StringValues block
+		sstring := ``
+		for _, k := range tmp {
+			sstring += `	` + k + ` = "` + p.StringValues[k] + `"
+`
+		}
+		s += sstring
+	}
+	// check IntValues not empty
+	if len(p.IntValues) != 0 {
+		// sort IntValues keys
+		tmp := make([]string, 0, len(p.IntValues))
+		for k := range p.IntValues {
+			tmp = append(tmp, k)
+		}
+		sort.Strings(tmp)
+
+		// create IntValues block
+		sint := ``
+		for _, k := range tmp {
```

Comment:
Do you want to try https://github.com/CleverCloud/terraform-provider-clevercloud/blob/master/pkg/map.go#L49 ?